crates/flycheck/src/lib.rs

```diff
@@ -258,53 +250,36 @@ impl FlycheckActor {
 }
 
 struct CargoHandle {
-    child: JodChild,
```

Comment:
@Veykril this removal of `JodChild` feels very suspicious. The job of `JodChild` is to kill the underlying `cargo` process,  when the `CargoHandle` is dropped. 

Notably, when we receive a `Restart` event from the caller, we want to cancel the currently-running `cargo check`. We can't just directly cancel it (there's no way to break out of `read` syscall nicely), but, if we kill the whole process (which we can do from a different thread). 

If we no longer want to terminate the process early, I think we can save one thread here (IIRC, the reason why we need both CargoHandle and FlycheckHandle is exactly to allow killing the process).

Reply:
Note that the previous logic was slightly suboptimal. We want `self.cargo_handle.take()` just before `while let Ok(Restart)` loop.

Reply:
Yeah, `` (drop) is not the most readable way to spell important stuff